### ISSUE TITLE
Improve replacement suggestions

### ISSUE THREAD

**Issue by jceddy** [jceddy/DailyArenaDeckAdvisor#8]
Improve replacement suggestions that are currently based on a fix per-card rank score.

Some ideas (might end up with some combination of these):
* Generate a score based based on how often a card shows up in decks with another card (i.e. it's more likely that a card will be selected as a replacement suggestion if it shows up in other deck lists that contain similar cards as the decklist being processed).
* Create a better scoring mechanism that takes more context into account. For example, most cards still have a fixed score, but some cards have multiple scores that are contingent on things like "5+ sacrifice outlets in the deck" (makes Act of Treason's score go up, for instance)...this requires some way to tag cards with tags like "Sacrifice Outlet". - This could lead to a more interactive deck builder functionality down the road, as well.
* Add some functionality for users to reject specific suggestions and get new ones.
* Allow users access to manually override the card scores.
* Improve functionality for suggesting cards with higher CMC/adjust lands as needed (HARD!)

**Comment by jceddy** [comment]
Pinning this, since it's the most up-voted next piece of functionality.

I'm wondering if anyone has anyone has input on what methods on the list above might work best, ideas on an alorithm/implementation to use, or even just other ideas on how to improve the suggestions.

**Comment by jceddy** [comment]
I added some code to track "card association" statistics, and use those to influence replacement suggestions. The effect it has on replacement suggestions is subtle, but the differences I was able to see seemed to make sense. (Commit https://github.com/jceddy/DailyArenaDeckAdvisor/commit/deb0de120127cbb5d908372ad0e715bb1e346194)

Here are some example differences:

For the Izzet Saheeli (Standard) deck, the following replacement suggestion

2x Disdainful Stroke => Anticipate

became

2x Disdainful Stroke => Negate

Seems legitimate, as you're now replacing a counterspell with a counterspell.

For the RDW (Arena Standard) deck, the following replacement suggestion

4x Cavalcade of Calamity => Lava Coil

became

3x Cavalcade of Calamity => Lightning Strike
1x Cavalcade of Calamity => Lava Coil

Basically, the algorithm now favors burn that can go to face, likely due to the association with Chandra's Spitfire, which has high synergy with noncombat damage to the opponent.

For the Liliana, Dreadhorde Geneneral (Brawl) deck, the following replacement suggestion

1x Liliana's Triumph => Unlikely Aid

becomes

1x Liliana's Triumph => Bladebrand

This also makes sense, since the deck has graveyard and death trigger synergies and should generally prefer trading your creatures with opponent creatures (and getting an extra card draw!) over saving your creature (even if it means your creature eats theirs).

If anyone has any input/comments on this, I'd be glad to see them. :)

**Comment by jceddy** [comment]
Somewhat related to this, I have now added code to enforce color identity when generating suggestions for Brawl decks. This required pulling in color identity information, which I hadn't been doing previously, and actually caring about which card the commander is for the deck. That information (commander card and color identity) is now shown separated on the archetype detail view for Brawl decks.

**Comment by jceddy** [comment]
Closing this issue for now, we can open a new one if we decide to do further rounds of improvements to the replacement suggestion algorithm.
